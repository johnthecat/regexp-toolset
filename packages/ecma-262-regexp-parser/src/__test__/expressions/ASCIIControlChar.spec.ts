import { expect, describe, it } from 'vitest';
import { parseRegexp } from '../../index.js';

/**
 * @see parseASCIIControlChar
 */
describe('ASCII Control Character', () => {
  it('should correctly parse control character', () => {
    expect(parseRegexp(/\cA/)).toMatchSnapshot('A');
    expect(parseRegexp(/\ca/)).toMatchSnapshot('a');
    expect(parseRegexp(/\cZ/)).toMatchSnapshot('Z');
    expect(parseRegexp(/\cz/)).toMatchSnapshot('z');
  });

  it('should correctly parse as escaped character at the end of the line', () => {
    expect(parseRegexp(/a\c/)).toMatchSnapshot();
  });

  it('should throw, if there is invalid character', () => {
    expect(() => parseRegexp(/\c1/)).toThrowErrorMatchingInlineSnapshot(`
      "
       ❱ /\\\\c1/
          ═══
       Invalid control character"
    `);
  });
});
