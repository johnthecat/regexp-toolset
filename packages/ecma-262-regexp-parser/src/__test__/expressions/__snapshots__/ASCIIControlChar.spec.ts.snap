// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ASCII Control Character > should correctly parse as escaped character at the end of the line 1`] = `
{
  "body": {
    "end": 3,
    "expressions": [
      {
        "charCode": 97,
        "end": 1,
        "kind": 6,
        "start": 1,
        "type": "simple",
        "value": "a",
      },
      {
        "charCode": 99,
        "end": 3,
        "kind": 6,
        "start": 2,
        "type": "escaped",
        "value": "c",
      },
    ],
    "kind": 5,
    "start": 1,
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`ASCII Control Character > should correctly parse control character > A 1`] = `
{
  "body": {
    "end": 3,
    "kind": 7,
    "start": 1,
    "value": "A",
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`ASCII Control Character > should correctly parse control character > Z 1`] = `
{
  "body": {
    "end": 3,
    "kind": 7,
    "start": 1,
    "value": "Z",
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`ASCII Control Character > should correctly parse control character > a 1`] = `
{
  "body": {
    "end": 3,
    "kind": 7,
    "start": 1,
    "value": "a",
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;

exports[`ASCII Control Character > should correctly parse control character > z 1`] = `
{
  "body": {
    "end": 3,
    "kind": 7,
    "start": 1,
    "value": "z",
  },
  "end": 4,
  "flags": "",
  "kind": 0,
  "start": 0,
}
`;
